Added FootballTeamGenerator v2.0

todorzag opened the pull request.

todorzag committed:
Added FootballTeamGenerator v2.0

mvlt commented on FootballTeamGenerators/FootballTeamGenerator v2.0/FootballTeamGenerator/Program.cs:
```diff
@@ -0,0 +1,101 @@
+
+                try
+                {
+                    if (action == "Team")
```
Why action "Team" is outside the switch

mvlt commented on FootballTeamGenerators/FootballTeamGenerator v2.0/FootballTeamGenerator/Program.cs:
```diff
@@ -0,0 +1,101 @@
+                    switch (action)
+                    {
+                        case "Add":
+                            Player player = CreatePlayerFromInput(input);
```
Wrap each action logic in a single method

mvlt commented on FootballTeamGenerators/FootballTeamGenerator v2.0/FootballTeamGenerator/Player.cs:
```diff
@@ -0,0 +1,115 @@
+                decimal.Parse(property.GetValue(this, null).ToString()));
+        }
+
+        private int StatHandler(int stat, string name)
```
Find more proper name. Should this function throw an exception like in Name

mvlt commented on FootballTeamGenerators/FootballTeamGenerator v2.0/FootballTeamGenerator/Player.cs:
```diff
@@ -0,0 +1,115 @@
+
+            init
+            {
+                if (value == " " || value == null)
```
Could we wrap this in a separate function

mvlt commented on FootballTeamGenerators/FootballTeamGenerator v2.0/FootballTeamGenerator/Player.cs:
```diff
@@ -0,0 +1,115 @@
+            Shooting = shooting;
+        }
+
+        private decimal CalculateOverall()
```
Endurance + Sprint + Dribble + Passing + Shooting /5 isn't this the logic

mvlt commented on FootballTeamGenerators/FootballTeamGenerator v2.0/FootballTeamGenerator/Team.cs:
```diff
@@ -0,0 +1,65 @@
+            _players.Remove(_players.Find(player => player.Name == playerName));
+        
+
+        public void CheckIfPlayerInTeam(string playerName, string teamName)
```
What do you expect CheckIfPlayerInTeam to do and return

mvlt commented on FootballTeamGenerators/FootballTeamGenerator v2.0/FootballTeamGenerator/Player.cs:
```diff
@@ -0,0 +1,115 @@
+    public class Player
+    {
+        private static List<string> restrictedProperties =
+            new List<string>
```
What will happen if we to add the age or some other property

todorzag committed:
Changes made:
Player:
 - Extracted init from property Name to Method
 - Renamed StatHandler Method to ValidateStat
Program:
 - Added comment for action Team
 - Extracted switch logic into separate methods
PlayerTests:
 - Remade tests to work with new logic

todorzag committed:
Changes Made:
Player:
 - Remade CalculateOverall logic
Team:
 - Extracted init into method
Tests:
 - Remade tests to work with new logic
Program:
 - Remade program to work with teams Dictionary

todorzag committed:
Remade Tests With NUnit and Remade Validate methods

todorzag committed:
Added CalculateRating_TeamIsValid_ReturnsAverageOfPlayerOverall test

todorzag committed:
Made NameTest into one Test
